virttest/utils_numeric.py

```diff
@@ -37,3 +38,40 @@ def format_size_human_readable(value, binary=False, precision='%.2f'):
     value = value * base / unit
     format_str = ('%d' if value.is_integer() else precision) + ' %s'
     return format_str % (value, s)
+
+
+def normalize_data_size(value_str, order_magnitude="M", factor="1024"):
+    """
+    Normalize a data size in one order of magnitude to another (MB to GB,
+    for example).
+
+    :param value_str: a string include the data default unit is 'B'
+    :param order_magnitude: the magnitude order of result
+    :param factor: the factor between two relative order of magnitude.
+                   Normally could be 1024 or 1000
+    """
+    def __get_unit_index(M):
+        try:
+            return ['B', 'K', 'M', 'G', 'T'].index(M.upper())
+        except ValueError:
+            pass
+        return 0
+
+    regex = r"(\d+\.?\d*)\s*(\w?)"
+    match = re.search(regex, value_str)
+    try:
+        value = match.group(1)
+        unit = match.group(2)
+        if not unit:
+            unit = 'B'
+    except TypeError:
+        raise ValueError("Invalid data size format 'value_str=%s'" % value_str)
+    from_index = __get_unit_index(unit)
+    to_index = __get_unit_index(order_magnitude)
+    scale = int(factor) ** (to_index - from_index)
+    data_size = float(value) / scale
+    # Control precision to avoid scientific notaion
+    if data_size.is_integer():
+        return "%.1f" % data_size
+    else:
+        return ("%.20f" % data_size).rstrip('0')
```

Comment:
`normalize_data_size('123921890', factor='1000')` will get result as `123.9218900000000047612`.
Redundant scale will lead to precision loss, will it be better to use decimal.Decimal?